apps/openmw/mwmechanics/actors.cpp

```diff
@@ -415,20 +415,24 @@ namespace MWMechanics
                 getActorsSidingWith(actor2, allies2, cachedAllies);
 
                 // Check that an ally of actor2 is also in combat with actor1
-                for (std::set<MWWorld::Ptr>::const_iterator it = allies2.begin(); it != allies2.end(); ++it)
+                for (const MWWorld::Ptr &ally2 : allies2)
                 {
-                    if ((it)->getClass().getCreatureStats(*it).getAiSequence().isInCombat(actor1))
+                    if (ally2.getClass().getCreatureStats(ally2).getAiSequence().isInCombat(actor1))
                     {
                         MWBase::Environment::get().getMechanicsManager()->startCombat(actor1, actor2);
                         // Also have actor1's allies start combat
-                        for (std::set<MWWorld::Ptr>::const_iterator it2 = allies1.begin(); it2 != allies1.end(); ++it2)
-                            MWBase::Environment::get().getMechanicsManager()->startCombat(*it2, actor2);
+                        for (const MWWorld::Ptr ally1 : allies1)
+                            MWBase::Environment::get().getMechanicsManager()->startCombat(ally1, actor2);
                         return;
                     }
                 }
             }
         }
 
+        // Stop here if target is unreachable
+        if (!canFight(actor1, actor2))
+            return;
```

Comment:
Is this reasonable to move before the followers are told to attack? If the player can't attack, but the follower can, this would stop them.

Reply:
Valid notion.